internal/test/mock/grpcclient.go

```diff
@@ -0,0 +1,131 @@
+package mock
+
+import (
+	"context"
+	"github.com/drand/drand/common"
+	"github.com/drand/drand/common/chain"
+	"github.com/drand/drand/common/client"
+	"github.com/drand/drand/protobuf/drand"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/peer"
+	"net"
+	"time"
+)
+
+type GrpcClient struct {
+	s *Server
+}
+
+func NewGrpcClient(s *Server) *GrpcClient {
+	return &GrpcClient{s: s}
+}
+
+func (c *GrpcClient) Get(ctx context.Context, round uint64) (client.Result, error) {
+	return c.s.PublicRand(ctx, &drand.PublicRandRequest{
+		Round:    round,
+		Metadata: nil,
+	})
+}
+
+func (c *GrpcClient) Watch(ctx context.Context) <-chan client.Result {
+	proxy := newStreamProxy(ctx)
+	go func() {
+		err := c.s.PublicRandStream(&drand.PublicRandRequest{}, proxy)
+		if err != nil {
+			proxy.Close()
+		}
+	}()
+	return proxy.outgoing
```

Comment:
I wonder if exposing the outgoing channel without the surrounding `proxy` could give resource leaks because `.Close()` is never called

Reply:
could possibly keep a list of these on the `GrpcClient` itself and close them in `Close()`